numba/cuda/mathimpl.py

```diff
@@ -0,0 +1,246 @@
+    def lower_boolean_impl(context, builder, sig, args):
+        libfunc_impl = context.get_function(libfunc,
+                                            typing.signature(types.int32, ty))
+        result = libfunc_impl(builder, args)
+        return context.cast(builder, result, types.int32, types.boolean)
+
+    lower(key, ty)(lower_boolean_impl)
+
+
+def impl_unary(key, ty, libfunc):
+    def lower_unary_impl(context, builder, sig, args):
+        libfunc_impl = context.get_function(libfunc, typing.signature(ty, ty))
+        return libfunc_impl(builder, args)
+
+    lower(key, ty)(lower_unary_impl)
+
+
+def impl_unary_int(key, ty, libfunc):
+    def lower_unary_int_impl(context, builder, sig, args):
+        if sig.args[0] == int64:
+            convert = builder.sitofp
+        else:
+            convert = builder.uitofp
+        arg = convert(args[0], ir.DoubleType())
+        sig = typing.signature(float64, float64)
+        libfunc_impl = context.get_function(libfunc, sig)
+        return libfunc_impl(builder, [arg])
+
+    lower(key, ty)(lower_unary_int_impl)
+
+
+def impl_binary(key, ty, libfunc):
+    def lower_binary_impl(context, builder, sig, args):
+        libfunc_impl = context.get_function(libfunc,
+                                            typing.signature(ty, ty, ty))
+        return libfunc_impl(builder, args)
+
+    lower(key, ty, ty)(lower_binary_impl)
+
+
+def impl_binary_int(key, ty, libfunc):
+    def lower_binary_int_impl(context, builder, sig, args):
+        if sig.args[0] == int64:
+            convert = builder.sitofp
+        else:
+            convert = builder.uitofp
+        args = [convert(arg, ir.DoubleType()) for arg in args]
+        sig = typing.signature(float64, float64, float64)
+        libfunc_impl = context.get_function(libfunc, sig)
+        return libfunc_impl(builder, args)
+
+    lower(key, ty, ty)(lower_binary_int_impl)
+
+
+for fname64, fname32, key in booleans:
+    impl32 = getattr(libdevice, fname32)
+    impl64 = getattr(libdevice, fname64)
+    impl_boolean(key, float32, impl32)
+    impl_boolean(key, float64, impl64)
+
+
+for fname64, fname32, key in unarys:
+    impl32 = getattr(libdevice, fname32)
+    impl64 = getattr(libdevice, fname64)
+    impl_unary(key, float32, impl32)
+    impl_unary(key, float64, impl64)
+    impl_unary_int(key, int64, impl64)
+    impl_unary_int(key, uint64, impl64)
+
+
+for fname64, fname32, key in binarys:
+    impl32 = getattr(libdevice, fname32)
+    impl64 = getattr(libdevice, fname64)
+    impl_binary(key, float32, impl32)
+    impl_binary(key, float64, impl64)
+    impl_binary_int(key, int64, impl64)
+    impl_binary_int(key, uint64, impl64)
+
+
+def impl_pow(ty, libfunc):
+    def lower_pow_impl(context, builder, sig, args):
+        pow_sig = typing.signature(ty, ty, ty)
+        libfunc_impl = context.get_function(libfunc, pow_sig)
+        return libfunc_impl(builder, args)
+
+    lower(math.pow, ty, types.int32)(lower_pow_impl)
```

Comment:
```suggestion
    lower(math.pow, ty, ty)(lower_pow_impl)
```
should it be that ^?

Reply:
The implementation was redundant, covered by `lower_binary_impl` from the list of `binarys` above: https://github.com/numba/numba/pull/6152/files#diff-3211256888268b006cb02721c73fd0807f610c0ef182d4a9aaf32cfcebd84f3fR48